### PR TITLE
Add project delete challenge

### LINKED ISSUE #285: Fixing permissions in connectors

### PR DESCRIPTION
**Pull request by thenav56**
Fix #285 #278 #286

* Fix permissions in connectors
* Add project delete challenge in project panel
* Fix redux state for project membership delete

Merge Dependent: https://github.com/toggle-corp/react-store/pull/384

### CODE REVIEW

File: src/redux/reducers/domainData/projects.js [hunk 1]
```diff
@@ -74,6 +77,38 @@ const emptyObject = {};
 
 // REDUCER
 
+const setUserProjectTitle = (state, action) => {
+    const {
+        project: { faramValues },
+        projectId,
+    } = action;
+    // This are the projectMiniUrlFields
```
Comment (tnagorra) [inline src/redux/reducers/domainData/projects.js]: "These are" should be the correct term :smiley:

File: src/views/Project/Details/General/Dashboard/index.js [hunk 2]
```diff
@@ -213,7 +213,7 @@ export default class ProjectDashboard extends React.PureComponent {
     }
 
     renderSourcers = () => {
-        const { projectDashboard: { topSourcers } } = this.props;
+        const { projectDashboard: { topSourcers = [] } } = this.props;
```
Comment (tnagorra) [inline src/views/Project/Details/General/Dashboard/index.js]: should use emptyList instead on instantiating new list everytime  on render

File: src/views/Project/Details/General/Dashboard/index.js [hunk 3]
```diff
@@ -231,7 +231,7 @@ export default class ProjectDashboard extends React.PureComponent {
     }
 
     renderTaggers = () => {
-        const { projectDashboard: { topTaggers } } = this.props;
+        const { projectDashboard: { topTaggers = [] } } = this.props;
```
Comment (tnagorra) [inline src/views/Project/Details/General/Dashboard/index.js]: should use emptyList instead on instantiating new list everytime  on render

File: src/views/Project/Details/General/Dashboard/index.js [hunk 4]
```diff
@@ -324,7 +324,7 @@ export default class ProjectDashboard extends React.PureComponent {
     )
 
     renderMap = () => {
-        const { projectDashboard: { regions } } = this.props;
+        const { projectDashboard: { regions = [] } } = this.props;
```
Comment (tnagorra) [inline src/views/Project/Details/General/Dashboard/index.js]: should use emptyList instead on instantiating new list everytime  on render

File: src/views/UserProfile/UserProject/index.js [hunk 5]
```diff
@@ -147,7 +144,14 @@ export default class UserProject extends React.PureComponent {
                     const activeUserMembership = (d.memberships || [])
                         .find(e => e.member === activeUser.userId);
 
-                    if (activeUserMembership && activeUserMembership.role === 'normal') {
+                    console.warn(activeUserMembership);
```
Comment (tnagorra) [inline src/views/UserProfile/UserProject/index.js]: we don't like console.warn :smile:

File: src/redux/reducers/domainData/projects.js (hunk 1 above)
Comment (thenav56) [reply src/redux/reducers/domainData/projects.js]: Done!

File: src/views/Project/Details/General/Dashboard/index.js (hunk 2 above)
Comment (thenav56) [reply src/views/Project/Details/General/Dashboard/index.js]: Done!

File: src/views/Project/Details/General/Dashboard/index.js (hunk 3 above)
Comment (thenav56) [reply src/views/Project/Details/General/Dashboard/index.js]: Done!

File: src/views/Project/Details/General/Dashboard/index.js (hunk 4 above)
Comment (thenav56) [reply src/views/Project/Details/General/Dashboard/index.js]: Done!

File: src/views/UserProfile/UserProject/index.js (hunk 5 above)
Comment (thenav56) [reply src/views/UserProfile/UserProject/index.js]: Done!

**Pull request merged**